doc/zettelkasten.txt

```diff
@@ -147,6 +151,20 @@ setup({opts})                                                *zettelkasten-setup
             - {id_pattern} (optional, string): The regex pattern used to
               determine the id of a note.
               Default value is: `%d+-%d+-%d+-%d+-%d+-%d+`
+            - {id_format} (optional, string, function): The format to use for
+              generateing new ids. If this is a string it is passed into
+              strftime to extract the current time in the specified format.
+              If this is a function it takes in the parent id passed to ZkNew
+              and the list of all the note ids, and should return a new id.
+              For example: >
+              {
+                'id_format': function(parent_id, ids)
```

Comment:
Passing in the IDs for all notes seem a bit excessive. If the function is interested in this, they can just use the public API and get the IDs themselves.

Reply:
That is fair. I have removed the parsing in of all the ids.